src/main/java/edu/ucsb/cs156/gauchoride/controllers/DriverAvailabilityController.java

```diff
@@ -0,0 +1,143 @@
+        Iterable<DriverAvailability> availabilities;
+
+        if (getCurrentUser().getRoles().contains(new SimpleGrantedAuthority("ROLE_ADMIN"))) {
+            availabilities = driverAvailabilityRepository.findAll();
+        } else {
+            availabilities = driverAvailabilityRepository.findAllByDriverId(getCurrentUser().getUser().getId());
+        }
+
+        return availabilities;
+    }
+
+    @Operation(summary = "Get a single availability by id, can only be current user's if not admin")
+    @PreAuthorize("hasRole('ROLE_ADMIN') || hasRole('ROLE_DRIVER')")
+    @GetMapping("")
+    public DriverAvailability getById(
+            @Parameter(name="id", description = "long, Id of the availability to get", 
+            required = true)  
+            @RequestParam Long id) {
+                DriverAvailability availability;
+        
+        if (getCurrentUser().getRoles().contains(new SimpleGrantedAuthority("ROLE_ADMIN"))) {
+            availability = driverAvailabilityRepository.findById(id)
+                .orElseThrow(() -> new EntityNotFoundException(DriverAvailability.class, id));;
+        } else {
+            availability = driverAvailabilityRepository.findByIdAndDriverId(id, getCurrentUser().getUser().getId())
+                .orElseThrow(() -> new EntityNotFoundException(DriverAvailability.class, id));
+        }
+
+        return availability;
+    }
+
+    @Operation(summary = "Create a new ride")
+    @PreAuthorize("hasRole('ROLE_ADMIN') || hasRole('ROLE_DRIVER')")
+    @PostMapping("/new")
+    public DriverAvailability postAvailability(
+        @Parameter(name="day", description="String, Day of the week ride is requested (Monday - Sunday) and allows Monday, Tuesday, Wednesday, Thursday, Friday, Saturday, Sunday", 
+                    example="Tuesday", required = true) 
+        @RequestParam String day,
+
+        @Parameter(name="startTime", description="String, Time the ride starts HH:MM(A/P)M", example="12:30AM", required = true)
+        @RequestParam String startTime,
+
+        @Parameter(name="endTime", description="String, Time the ride ends HH:MM(A/P)M", example="12:30AM", required = true)
+        @RequestParam String endTime,
+
+         @Parameter(name="notes", description="String, Extra information", example="We have two people riding", required = true)
+        @RequestParam String notes
+        )
+        {
+
+        DriverAvailability availability = new DriverAvailability();
+        
+        availability.setDriverId(getCurrentUser().getUser().getId());
+        availability.setDay(day);
+        availability.setStartTime(startTime);
+        availability.setEndTime(endTime);
+        availability.setNotes(notes);
+
+        DriverAvailability savedAvailability = driverAvailabilityRepository.save(availability);
+
+        return savedAvailability;
+    }
+
+    @Operation(summary = "Delete an availability if owned by current user")
```

Comment:
Admin should also be able to delete anyones availability, sorry for being unclear in the issue description